utils/analyticsmetrics.go

```diff
@@ -0,0 +1,156 @@
+	xscservices "github.com/jfrog/jfrog-client-go/xsc/services"
+	"os"
+	"strings"
+	"time"
+)
+
+const AnalyticsMetricsMinXscVersion = "1.7.1"
+
+type AnalyticsMetricsService struct {
+	xscManager *xsc.XscServicesManager
+	// Should the CLI reports analytics metrics to XSC.
+	shouldReportEvents bool
+	msi                string
+	startTime          time.Time
+}
+
+func NewAnalyticsMetricsService(serviceDetails *config.ServerDetails) *AnalyticsMetricsService {
+	ams := AnalyticsMetricsService{}
+	xscManager, err := CreateXscServiceManager(serviceDetails)
+	if err != nil {
+		// When an error occurs, shouldReportEvents will be false and no XscServiceManager commands will be executed.
+		log.Debug(fmt.Sprintf("Failed to create xsc manager for analytics metrics service. %s", err.Error()))
+		return &ams
+	}
+	ams.xscManager = xscManager
+	ams.shouldReportEvents = ams.calcShouldReportEvents()
+	return &ams
+}
+
+func (ams *AnalyticsMetricsService) calcShouldReportEvents() bool {
+	// A user who explicitly requests not to send reports will not receive XSC analytics metrics.
+	if os.Getenv(coreutils.ReportUsage) == "false" {
+		return false
+	}
+	// There is no need to report the event and generate a new msi for the cli scan if the msi was provided.
+	if os.Getenv(jfMsiEnvVariable) != "" {
+		return false
+	}
+	// Verify xsc version.
+	xscVersion, err := ams.xscManager.GetVersion()
+	if err != nil {
+		return false
+	}
+	if err = clientutils.ValidateMinimumVersion(clientutils.Xsc, xscVersion, AnalyticsMetricsMinXscVersion); err != nil {
+		return false
+	}
+	return true
+}
+
+func (ams *AnalyticsMetricsService) SetMsi(msi string) {
+	ams.msi = msi
+}
+
+func (ams *AnalyticsMetricsService) GetMsi() string {
+	return ams.msi
+}
+
+func (ams *AnalyticsMetricsService) SetStartTime() {
+	ams.startTime = time.Now()
+}
+
+func (ams *AnalyticsMetricsService) GetStartTime() time.Time {
+	return ams.startTime
+}
+
+func (ams *AnalyticsMetricsService) ShouldReportEvents() bool {
+	return ams.shouldReportEvents
+}
+
+func (ams *AnalyticsMetricsService) AddGeneralEventAndSetMsi(params *services.XrayGraphScanParams) {
+	if !ams.ShouldReportEvents() {
+		log.Debug("A general event request was not sent to XSC - analytics metrics are disabled.")
+		return
+	}
+	err := ams.AddGeneralEvent()
+	if err != nil {
+		log.Debug(fmt.Errorf("failed sending general event request to XSC service, error: %s ", err.Error()))
+		return
+	}
+	log.Debug(fmt.Sprintf("New General event added successfully. multi_scan_id %s", ams.GetMsi()))
+
+	if err = os.Setenv(jfMsiEnvVariable, ams.GetMsi()); err != nil {
+		// Not a fatal error, if not set the scan will not be shown at the XSC UI, should not fail the scan.
+		log.Debug(fmt.Sprintf("failed setting MSI as environment variable. Cause: %s", err.Error()))
+	}
+	// Before running the audit command, set the msi so the sca scan will be performed on the xsc rather than on the xray server.
+	params.MultiScanId = ams.GetMsi()
```

Comment:
This method should return the msi and it should be set at the AuditBasicParams (that will set the value of XrayGraphScanParams)
This env var is used to pass the value to the analyzer manager only